src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/GovUKHolidayClient.kt

```diff
@@ -0,0 +1,54 @@
+package uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.client
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.cache.annotation.Cacheable
+import org.springframework.core.io.ClassPathResource
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.bodyToMono
+import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.govuk.holidays.HolidaysDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.service.GovUkHolidayService
+import java.nio.file.Files
+import java.nio.file.Path
+import java.time.Duration
+
+@Component
+class GovUKHolidayClient(
+  @param:Qualifier("govUKWebClient") private val govUKWebClient: WebClient,
+  @param:Value("\${gov-uk.api.timeout:10s}") val apiTimeout: Duration,
+
+  @param:Value("\${gov-uk.api.bank-holidays-local-cache}")
+  private val localCacheResourceFile: String,
+
+  private val mapper: ObjectMapper = ObjectMapper().registerModule(KotlinModule.Builder().build())
+    .registerModule(JavaTimeModule()),
+) {
+  companion object {
+    val LOG: Logger = LoggerFactory.getLogger(this::class.java)
+    const val HOLIDAYS_JSON = "bank-holidays1.json"
+  }
+
+  @Cacheable("bank-holidays", unless = "#result == null")
```

Comment:
(Kristian) That bank holiday “API” is a json file that doesn’t change much. Suggest it’s implemented with a 7 day cache (or risk being throttled by that service).

Added caching based on the ticket. Needs to be discussed though